### ISSUE TITLE
jsdoc/require-jsdoc false positive with react typescript (tsx) files

### ISSUE THREAD

**Issue by mauro0x52** [gajus/eslint-plugin-jsdoc#445]
### Summary

I'm having some troubles trying to use eslint-plugin-jsdoc with react typescript files. I get a `Missing JSDoc comment.eslint(jsdoc/require-jsdoc)`, even though there is a jsdoc there.

### How to reproduce:

* Create a renderer method that returns a self closing tag
* Create a method right after the rendered method

Environment:
- node 12.13.1
- eslint 6.4.0
- eslint-plugin-jsdoc 18.4.3

Configuration:
```
const config = {
  parserOptions: {
    ecmaVersion: 2019,
    sourceType: 'module',
    ecmaFeatures: {
      jsx: true,
      impliedStrict: true
    }
  },
  parser: '@typescript-eslint/parser',
  plugins: [
    'jsdoc'
  ],
  rules: {
    'jsdoc/require-jsdoc': [
      'error',
      {
        require: {
          FunctionDeclaration: true,
          FunctionExpression: true,
          ClassDeclaration: true,
          ClassExpression: true,
          MethodDefinition: true
        }
      }
    ]
  }
};
```

### Example:

```
import React from 'react';
import { View } from 'react-native';

/**
 * Example component
 */
class Example {
  /**
   * Renders a single item
   * @returns the item react element
   */
  renderItem(): React.ReactElement {
    return <View style={{ position: 'relative' }} />;
  }

  /**
   * Renders a list of items
   * @returns the list react element
   */
  renderList(): React.ReactElement { // here complains "Missing JSDoc comment.eslint(jsdoc/require-jsdoc)"
    return <View />;
  }
}
```

### Workaround

Transforming the tag into non-self-closing works.

```
import React from 'react';
import { View } from 'react-native';

/**
 * Example component
 */
class Example {
  /**
   * Renders a single item
   * @returns the item react element
   */
  renderItem(): React.ReactElement {
    return <View style={{ position: 'relative' }}></View>;
  }

  /**
   * Renders a list of items
   * @returns the list react element
   */
  renderList(): React.ReactElement { // no more complains
    return <View />;
  }
}
```

**Comment by brettz9** [comment]
I don't know what your environment is, but when I add the following settings to the `require-jsdoc` rules within our testing suite (at `test/rules/assertions/requireJsdoc ), it passed under "valid" code:

```js
{
    code: `
    import React from 'react';
    import { View } from 'react-native';

    /**
     * Example component
     */
    class Example {
      /**
       * Renders a single item
       * @returns the item react element
       */
      renderItem(): React.ReactElement {
        return <View style={{ position: 'relative' }} />;
      }

      /**
       * Renders a list of items
       * @returns the list react element
       */
      renderList(): React.ReactElement { // here complains "Missing JSDoc comment.eslint(jsdoc/require-jsdoc)"
        return <View />;
      }
    }
    `,
    parser: require.resolve('@typescript-eslint/parser'),
    parserOptions: {
      ecmaFeatures: {
        jsx: true,
      },
      ecmaVersion: 6,
      sourceType: 'module',
    },
  },
```

**Comment by mauro0x52** [comment]
Hello @brettz9.

My bad, I forgot to add more details. I also updated the description above just in case.

- node 12.13.1
- eslint 6.4.0
- eslint-plugin-jsdoc 18.4.3

The configuration I'm running:
```
const config = {
  parserOptions: {
    ecmaVersion: 2019,
    sourceType: 'module',
    ecmaFeatures: {
      jsx: true,
      impliedStrict: true
    }
  },
  parser: '@typescript-eslint/parser',
  plugins: [
    'jsdoc'
  ],
  rules: {
    'jsdoc/require-jsdoc': [
      'error',
      {
        require: {
          FunctionDeclaration: true,
          FunctionExpression: true,
          ClassDeclaration: true,
          ClassExpression: true,
          MethodDefinition: true
        }
      }
    ]
  }
};
```

**Comment by brettz9** [comment]
I am not able to replicate even with that config and those options. Maybe you could produce a minimal test repo?

**Comment by mauro0x52** [comment]
Hi @brettz9 !

I found out the version of the @typescript-eslint/parser I was using was outdated. Updating it made it work.

Thanks for your support!
